presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1062,17 +1062,24 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
     {
         HiveTableLayoutHandle hiveLayoutHandle = checkType(layoutHandle, HiveTableLayoutHandle.class, "layoutHandle");
         List<ColumnHandle> partitionColumns = hiveLayoutHandle.getPartitionColumns();
-        List<TupleDomain<ColumnHandle>> partitionDomains = hiveLayoutHandle.getPartitions().get().stream()
-                .map(HivePartition::getTupleDomain)
-                .collect(toList());
+        List<HivePartition> partitions = hiveLayoutHandle.getPartitions().get();
 
-        TupleDomain<ColumnHandle> predicate = TupleDomain.none();
-        if (!partitionDomains.isEmpty()) {
-            predicate = TupleDomain.columnWiseUnion(partitionDomains);
```

Comment:
Now I see the usage.  In the short term, we can limit this to when partition count exceeds some limit.  In the long term we need an optimize version of 
```java
TupleDomain.columnWiseUnion(hiveLayoutHandle.getPartitions().get().stream()
                .map(HivePartition::getTupleDomain)
                .collect(toList()));
```
That does not require creating a tuple domain for every partition as this is super expensive for large tables